### ISSUE TITLE
Test coverage support

### ISSUE THREAD

**Issue by bergie** [flowbased/fbp-spec#129]
It would be nice to be able to get test coverage reports from fbp-spec. Something like https://www.npmjs.com/package/nyc

**Comment by bergie** [comment]
Note: nyc claims suppor for _applications that spawn subprocesses_ so fbp-spec might be possible to make work.

**Comment by jonnor** [comment]
Does `nyc` produce a report file somewhere from running the tests? Have you tried wrapping the runtime command in it to see if it just works?

**Comment by jonnor** [comment]
Generally this is something that needs to happen on the runtime side, and in a runtime-specific manner, as how to compute coverage depends entirely on the language/execution-environment in use.

**Comment by bergie** [comment]
@jonnor tried wrapping fbp-spec (via its mocha interface) to `nyc`, and at least that didn't work:

https://github.com/noflo/noflo-core/pull/108

**Comment by jonnor** [comment]
@bergie what do you mean does not work? Looks fine here

![noflo-core-nyc-coverage](https://user-images.githubusercontent.com/45185/32180638-c8f150f2-bd92-11e7-9d59-e6ac3e3620b7.png)

**Comment by bergie** [comment]
@jonnor yes, see zero coverage reported for `Repeat` and `RepeatAsync`, which are fbp-spec tested

**Comment by jonnor** [comment]
@bergie ok. Does not work when running nyc infront of `fbp-spec` either, does not pick up any files at all.
So either subprocess tracking is off, it cannot handle the indirect file loading (though noflo.ComponentLoader seems to work for other), there is a problem with .js versus .coffee files or similar

**Comment by bergie** [comment]
@jonnor for `.coffee` files you will need to require [coffee-coverage](https://github.com/benbria/coffee-coverage)

See https://github.com/benbria/coffee-coverage/blob/master/docs/HOWTO-nyc.md

**Comment by jonnor** [comment]
Presumably that needs to be done in the process which uses the coffee files. Which would explain why it is not working in fbp-spec case: It is the runtime that requires the files, and there is no such registration there. `noflo-nodejs` does not have such an option either, I think?

**Comment by bergie** [comment]
@jonnor correct, we should probably add a mocha-style `--require` option to noflo-nodejs.

But first to verify this, it might make sense to add this manually to noflo-nodejs
